src/Avalonia.Diagnostics/Diagnostics/DevTools.cs

```diff
@@ -38,15 +46,21 @@ public static IDisposable Open(TopLevel root)
             {
                 window = new MainWindow
                 {
-                    Width = 1024,
-                    Height = 512,
+                    Width = options.Size.Width,
+                    Height = options.Size.Height,
                     Root = root,
                 };
 
                 window.Closed += DevToolsClosed;
+                window.ViewModel.ShouldVisualizeDirtyRects = options.ShouldVisualizeDirtyRects;
+                window.ViewModel.ShouldVisualizeMarginPadding = options.ShouldVisualizeMarginPadding;
+                window.ViewModel.Console.IsVisible = options.ShowConsole;
+                window.ViewModel.ShowFpsOverlay = options.ShowFpsOverlay;
+                window.ViewModel.ShowLayoutVisualizer = options.ShowLayoutVisualizer;                
+                window.ViewModel.ShowFpsOverlay = options.ShowFpsOverlay;
                 s_open.Add(root, window);
 
-                if (root is Window inspectedWindow)
+                if (options.OnTop && root is Window inspectedWindow)
```

Comment:
Not sure `OnTop` is the correct terminology here: "Topmost" windows have a specific meaning and but this switch isn't controlling that - it's controlling whether DevTools is shown as a child window.